compiler/noirc_frontend/src/utils.rs

```diff
@@ -0,0 +1,17 @@
+use acvm::{AcirField, FieldElement};
+use num_bigint::{BigInt, BigUint, Sign};
+
+pub fn big_int_to_field_element(big_int: &BigInt) -> FieldElement {
+    let big_uint = big_int.magnitude();
+    FieldElement::from_be_bytes_reduce(&big_uint.to_bytes_be())
+}
+
+pub fn field_element_to_big_int(field: &FieldElement) -> BigInt {
+    let big_uint = BigUint::from_bytes_be(&field.to_be_bytes());
+    BigInt::from_biguint(Sign::Plus, big_uint)
+}
+
+pub fn truncate_big_int_to_u128(big_int: &BigInt) -> u128 {
+    // TODO: use a more efficient way to do this
+    big_int_to_field_element(big_int).to_u128()
```

Comment:
I'm sure there's a better way to do this, probably iterating the be/le bytes... but I wanted to wait for a review before changing this.

Reply:
```rust
pub fn truncate_big_int_to_u128(big_int: &BigInt) -> u128 {
    let bigint_bytes = big_uint.to_bytes_le();
    let u128_bytes: [u8; 16] = bigint_bytes[0..16].try_into().unwrap();
    u128::from_le_bytes(u128_bytes)
}
```

This should work (up to dealing with endianness issues)

Reply:
Alternatively

```rust
pub fn truncate_big_int_to_u128(big_int: &BigInt) -> u128 {
    let bigint_u64_le_iter = big_uint.iter_u64_digits();
    let low = bigint_u64_le_iter.next().unwrap_or(0) as u128;
    let high = bigint_u64_le_iter.next().unwrap_or(0) as u128;
    high << 64 + low
}
```

Reply:
I tried both approached but they fail the test I just wrote. I'll use those as a base code, I'm sure it's almost there (but I'll switch to the lsp PR now)